src/content/app/genome-browser/components/track-panel/trackPanelConfig.ts

```diff
@@ -52,9 +55,19 @@ export type TrackStates = {
   };
 };
 
+// we might need to add other types for variants and regulation?
+export type GeneTrackStates = {
+  [objectId: string]: {
+    gene: {
+      status?: TrackActivityStatus;
+      transcripts?: string[];
+    };
+  };
+};
+
 export type GenomeTrackStates = {
   commonTracks?: TrackStates;
-  objectTracks?: { [objctId: string]: TrackStates };
+  objectTracks?: GeneTrackStates;
```

Comment:
We are probably looking for something like this instead:

```ts
type FocusTrack = FocusGeneTrack; // will be a union with FocusRegionTrack, FocusVariantTrack, etc.

type FocusGeneTrack = {
  type: 'gene';
  status: TrackActivityStatus;
  transcripts: string[] | null
};

type GenomeTrackStates = {
  objectTracks?: {
    [objectId: string]: FocusTrack;
  }
};
```

Reply:
✅